src/util/simplify_expr_pointer.cpp

```diff
@@ -410,7 +410,39 @@ static bool simple_structural_lvalue_equal(const exprt &lhs, const exprt &rhs)
   return false;
 }
 
-static exprt simplify_inequality_address_of_prepare_op(const exprt &op)
+// rewrite expression &a.b to &a when appropriate
```

Comment:
It may  be better to do this the other way around. Will add more info in the global comment.